### PR TITLE
Cairo 2.8.2

### PR DESCRIPTION
**Pull request by augustbleeds**
Note: **this will NOT be merged to `develop` until the audit is completed in order to avoid the situation where Ops deploys updated contracts.**

Bumping scarb/cairo to version 2.8.2. Because this changes the compile contracts (and the dependencies), this will stay as a PR and will not be merged. After the audit is completed, then it'll be merged to develop

 - [x] bump cairo/starknet version
 - [x]  open zeppelin bump -- make sure OZ is updated, but also bc there is a breaking erc-20 change for event indexing
 - [x] update build process (no longer use cairo directly, but via scarb)
 - [x] update foundry 
 - [x] add recently added contracts (MCM + Timelock)

### CODE REVIEW

File: .github/actions/install-cairo/action.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ runs:
       id: install-scarb
       shell: bash
       run: |
-        wget https://github.com/software-mansion/scarb/releases/download/${{ inputs.scarb_version }}/scarb-${{ inputs.scarb_version }}-x86_64-unknown-linux-musl.tar.gz
-        tar -xvzf scarb-${{ inputs.scarb_version }}-x86_64-unknown-linux-musl.tar.gz
-        mv -vf scarb-${{ inputs.scarb_version }}-x86_64-unknown-linux-musl scarb-build
+        wget https://github.com/software-mansion/scarb/releases/download/${{ inputs.scarb_version }}/scarb-${{ inputs.scarb_version }}-x86_64-unknown-linux-gnu.tar.gz
+        tar -xvzf scarb-${{ inputs.scarb_version }}-x86_64-unknown-linux-gnu.tar.gz
```
Comment (augustbleeds) [inline .github/actions/install-cairo/action.yml]: the musl version does not work anymore. gnu version does though

File: contracts/Scarb.toml [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
 [dependencies]
-starknet = ">=2.6.3"
-openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", tag = "v0.10.0" }
+starknet = "2.8.2"
+openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", tag = "v0.15.1" }
+alexandria_bytes = { git = "https://github.com/keep-starknet-strange/alexandria.git", rev = "e1b080577aaa6889116fc8be5dde72b2fd21e397" }
+alexandria_encoding = { git = "https://github.com/keep-starknet-strange/alexandria.git", rev = "e1b080577aaa6889116fc8be5dde72b2fd21e397" }
+
+[dev-dependencies]
+alexandria_math = { git = "https://github.com/keep-starknet-strange/alexandria.git", rev = "e1b080577aaa6889116fc8be5dde72b2fd21e397" }
```
Comment (archseer) [inline contracts/Scarb.toml]: Why the new dependencies?

Comment (augustbleeds) [reply contracts/Scarb.toml]: That's for MCMS. When I created this PR, MCMS wasn't fully merged to develop yet. Need to add them to this branch

File: contracts/src/access_control/rbac_timelock.cairo [hunk 3]
```diff
@@ -41,8 +40,8 @@ trait IRBACTimelock<TContractState> {
     fn update_delay(ref self: TContractState, new_delay: u256);
     fn block_function_selector(ref self: TContractState, selector: felt252);
     fn unblock_function_selector(ref self: TContractState, selector: felt252);
-    fn get_blocked_function_selector_count(self: @TContractState) -> u256;
-    fn get_blocked_function_selector_at(self: @TContractState, index: u256) -> felt252;
```
Comment (calvwang9) [inline contracts/src/access_control/rbac_timelock.cairo]: does this functionally change the interface, i.e. tooling needs updating?

**Review by calvwang9 — APPROVED** [review]
looks fine to me, although i dont have much context around cairo updates

**Pull request closed without merging**